pkgs/development/tools/phase/update.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl jq nix-prefetch common-updater-scripts nix coreutils
+# shellcheck shell=bash
+set -euo pipefail
+
+# Fetch the latest Phase CLI version from GitHub https://github.com/phasehq/cli
+REPO="phasehq/cli"
+RELEASE_TAG=$(curl -s "https://api.github.com/repos/$REPO/releases/latest" | jq -r .tag_name)
+VERSION=${RELEASE_TAG#v}
+
+echo "Latest Phase CLI release: $VERSION"
+
+# Prepare x86 and arm64 targets for Linux and MacOS
+ARCHS=(
+    "x86_64-linux:linux_amd64"
+    "x86_64-darwin:macos_amd64"
+    "aarch64-linux:linux_arm64"
+    "aarch64-darwin:macos_arm64"
+)
+
+# Paths to default.nix and SRI (Sub-resource integrity) file
+NIXCONFIG=pkgs/development/tools/phase/default.nix
+SRIFILE=pkgs/development/tools/phase/hashes.json
+
+# Clear file from a previous run
+rm -f "$SRIFILE" && touch "$SRIFILE"
+
+printf "{ \"_comment\": \"@generated by pkgs/development/tools/phase/update.sh\"\n" >> "$SRIFILE"
+
+# Download sha256sums for all targets from Phase CLI's GitHub releases
+for arch in "${ARCHS[@]}"; do
+    IFS=: read -r arch_name arch_target <<< "$arch"
+
+    # Get the SHA256 sum directly from the release assets
+    checksum_url="https://github.com/$REPO/releases/download/v${VERSION}/phase_cli_${arch_target}_${VERSION}.zip.sha256"
+    sha256sum=$(curl -sL "$checksum_url" | cut -d' ' -f1)
+
+    # Convert the SHA256 Hex hash to base64 for SRI
+    srihash="$(nix hash to-sri --type sha256 "$sha256sum")"
+    echo ", \"$arch_name\": \"$srihash\"" >> "$SRIFILE"
+done
+echo "}" >> "$SRIFILE"
+
+# Set the version in default.nix
+sed -i \
+  '0,/version\s*=\s*".*";/s//version = "'"$VERSION"'";/' \
+  "$NIXCONFIG"
```

Comment:
There's already a script in Nixpkgs called `update-source-version` that does this exact thing.